R/stats.R

```diff
@@ -22,23 +22,34 @@ n_complete <- function(x) {
   length(x) - n_missing(x)
 }
 
+
+#' Create a contingency table and arrange its levels in descending order
+#' 
+#' @param x An object that can be interpreted as a factor (including logical)
+#' @return A "table" object, which will be treated as a named numeric vector
+#' @export
+
+sorted_count <- function(x) {
+  tab <- table(x, useNA = "always")
+  out <- setNames(as.integer(tab), names(tab))
+  sort(out, decreasing = TRUE)
+}
+
 #' Generate inline histogram for numeric variables
 #' 
-#' @param x A vector
-#' @return A A numeric value of 0 with a name that is a character string of histogram.
+#' @param x A numeric vector
+#' @return A length-one character vector containing the histogram.
 #' @export
 
 inline_hist <- function(x) {
-  x <- x[!is.na(x)]
-  out <- 0
-  if ( !all(x == 0) & length(x) != 0) {
-    hist_dt <- table(cut(x, 10))
-    hist_dt <- hist_dt / max(hist_dt)
-    names(out) <- colformat::spark_bar(hist_dt)
-    return(out)
-  }
-  names(out) <- ""
-  return(out)
+  # Handle empty vectors
```

Comment:
I believe the NA handling was because at the unconf we found a big problem with all NA columns and this function.

Reply:
Not handled anywhere at this time but we probably want to have a maximum length here as we do with the line graph. Such as 40 characters or something like that.

Reply:
A later commit changes this to use the character formatting options within the package.

Reply:
Let's make sure to have a test for the all NA case (really we should for all of the types).